### PR TITLE
expose badger loading mode conf

### PR DESCRIPTION
**Pull request by lanzafame**
Resolves: #771.

For simplicity, I haven't exposed the entire
badger config. Also, due to go default values and
the badger implementation, I have had to extract the
two loading mode config values so we can determine
whether it has actually has been set by the user in
the json config file.

License: MIT
Signed-off-by: Adrian Lanzafame <adrianlanzafame92@gmail.com>

### CODE REVIEW

File: datastore/badger/config.go [hunk 1]
```diff
@@ -81,6 +112,12 @@ func (cfg *Config) LoadJSON(raw []byte) error {
 
 func (cfg *Config) applyJSONConfig(jcfg *jsonConfig) error {
 	config.SetIfNotDefault(jcfg.Folder, &cfg.Folder)
+	if v, ok := correctLoadingMode(jcfg.TableLoadingMode); ok {
+		cfg.BadgerOptions.TableLoadingMode = options.FileLoadingMode(v)
```
Comment (hsanjuan) [inline datastore/badger/config.go]: casting `options.FileLoadingMode` twice. Also, shouldn't it error ?

Comment (hsanjuan) [reply datastore/badger/config.go]: You can make `jsonConfig.TableLoadingMode` a pointer to decide whether it's set at all (with a possible wrong value) or not.

File: datastore/badger/config_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	"testing"
+
+	"github.com/dgraph-io/badger"
+	"github.com/dgraph-io/badger/options"
+)
+
+var cfgJSON = []byte(`
+{
+    "folder": "test",
+	"value_log_loading_mode": 1
```
Comment (hsanjuan) [inline datastore/badger/config_test.go]: indentation

File: go.mod [hunk 3]
```diff
@@ -10,7 +10,8 @@ require (
 	github.com/boltdb/bolt v1.3.1 // indirect
 	github.com/btcsuite/btcd v0.0.0-20190427004231-96897255fd17 // indirect
 	github.com/coreos/go-semver v0.3.0 // indirect
-	github.com/dgryski/go-farm v0.0.0-20190423205320-6a90982ecee2 // indirect
+	github.com/dgraph-io/badger v2.0.0-rc.2+incompatible
+	github.com/dgryski/go-farm v0.0.0-20190416075124-e1214b5e05dc // indirect
```
Comment (hsanjuan) [inline go.mod]: interesting, why is this one changing?

**Comment by lanzafame** [comment]
> I thought we decided to try reducing the value log size so that it fits in memory, and this it not exposing that at all ?

I tried that before making these changes but even significantly reduced it didn't work.

**Comment by hsanjuan** [comment]
> > I thought we decided to try reducing the value log size so that it fits in memory, and this it not exposing that at all ?
> 
> I tried that before making these changes but even significantly reduced it didn't work.

Can we expose it in any case, since we're at it?

File: go.mod (hunk 3 above)
Comment (lanzafame) [reply go.mod]: hmm not sure, I will check it out.

**Comment by lanzafame** [comment]
> > > I thought we decided to try reducing the value log size so that it fits in memory, and this it not exposing that at all ?
> > 
> > 
> > I tried that before making these changes but even significantly reduced it didn't work.
> 
> Can we expose it in any case, since we're at it?

That was painful but it is now exposing the full config now.

**Comment by hsanjuan** [comment]
agh I forgot to merge this

**Pull request merged**